### PR TITLE
Fixes #2172, add rel=nofollow in layered navigation links

### LINKED ISSUE #2172: Suggestion to add rel="nofollow" in layered navigation links to prevent index by GoogleBot.
**Is your feature request related to a problem? Please describe.**
If managing a large number of product attributes in Magento2,  GoogleBot will do a huge number of crawls by following the links in the layered navigation.

**Describe the solution you'd like**
Add rel="nofollow" to the navigation link following to the default template of Magento2(2.4.2).

Smile_ElasticsuiteCatalog/web/template/attribute-filter.html:12
`<a data-bind="attr: {href:url}, visible: count >= 1">`
↓
`<a data-bind="attr: {href:url}, visible: count >= 1" rel="nofollow">`

**Additional context**

Magento2(2.4.2) default template

magento/module-layered-navigation/view/frontend/templates/layer/filter.phtml:21
`<a href="<?= $escaper->escapeUrl($filterItem->getUrl()) ?>" rel="nofollow">`

### PR DESCRIPTION
**Pull request by vahonc**

### CODE REVIEW

**Comment by romainruaud** [comment]
@vahonc can you make it configurable ? It would be nice to have a switch for each attribute where user can select the value of "follow", "nofollow" for each attribute ?

**Comment by romainruaud** [comment]
To clarify, what I'd like to have is : 

In Back-Office : 

- a new field for each attribute edit screen.
- the field is named "Meta Robots" (meta_robots), that will have this options : https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Config/Model/Config/Source/Design/Robots.php
- the default value is "INDEX, FOLLOW".
- this field is displayed only if the attribute is "is_filterable" or "is_filterable_in_search"

In Front-Office : 
- apply the proper value to the html rendering from the attribute data (as it's done for "facet_max_size" or "facet_sort_order".

(Bonus) In GraphQL : 
- add a field "meta_robot" in the Aggregations, like it's done for the "has_more" here : https://github.com/Smile-SA/elasticsuite/commit/7e18e2504a08bee1b84a6e95f51e68bdb0e32866

File: src/module-elasticsuite-catalog/Setup/CatalogSetup.php [hunk 1]
```diff
@@ -583,6 +583,32 @@ public function addFilterBooleanLogicField(SchemaSetupInterface $setup)
...
+
+        // Append a column 'is_display_rel_nofollow' into the db.
+        $connection->addColumn(
+            $table,
+            'is_display_rel_nofollow',
+            [
+                'type'     => \Magento\Framework\DB\Ddl\Table::TYPE_BOOLEAN,
+                'nullable' => false,
+                'default'  => '0',
+                'comment'  => 'Boolean logic to use for displaying rel="nofollow" attribute for all filter links of current attribute',
```
Comment (romainruaud) [inline src/module-elasticsuite-catalog/Setup/CatalogSetup.php:606]: The comment is not accurate :) it looks like a copy-paste :)

**Review by romainruaud — CHANGES_REQUESTED** [review]

**Comment by romainruaud** [comment]
please fix the requested changes and also the PHPMD, and we'll be good to merge

**Comment by CalzaFino** [comment]
Can you please confirm if this feature was implemented? I did update to the newest version (2.10.11) but I can't find anywhere the menu to set the noindex. I do have several filtrable attributes and this option is not visible.
Thank you for all your hard work providing this module.

**Comment by vahonc** [comment]
Hello @CalzaFino,

In order to see this field, you need to set the value `Filterable (with results)` for the option `Use in Layered Navigation` on the `Storefront Properties` tab. And this is a prerequisite. 

Look at the screenshot below:

Screenshot from 2022-08-15 09-36-54

As you can see the field `Add rel="nofollow" to filter links in Layered Navigation`will be displayed right after the field `Use in Layered Navigation`.

BR,
Vadym

**Pull request merged**